packages/preview-middleware-client/src/cpe/utils.ts

```diff
@@ -112,3 +138,56 @@ export function getRootControlFromComponentContainer(container?: ComponentContai
     }
     return undefined;
 }
+
+export function getManifestProperties(
+    control: ManagedObject,
+    changeService: ChangeService,
+    controlOverlay?: ElementOverlay
+): {
+    [key: string]: MergedSetting;
+} {
+    const overlayData = controlOverlay?.getDesignTimeMetadata().getData();
+    if (!controlOverlay || !overlayData?.manifestSettings) {
+        return {};
+    }
+    const manifestPropertiesValue = overlayData?.manifestSettingsValues(
+        overlayData?.manifestSettings(control),
+        control
+    );
+    const manifestProperties = overlayData?.manifestSettings(control).reduce(
+        (
+            acc: {
+                [key: string]: MergedSetting;
+            },
+            item: DesigntimeSetting
+        ) => {
+            const propertyId = item.id;
+            const change = changeService.pendingChanges.filter(
```

Comment:
This change specific logic probably should be in the change service. What we are interested here is to get the configuration change value for a particular property and it seems quite generic use case, so the change service could provide API for that. Also storing values in a map on stack change event would probably be easier than looping over all changes for each property.

Reply:
Please review